spec/core/group_by_spec.rb

```diff
@@ -465,4 +465,21 @@
 
     it { is_expected.to eq Daru::DataFrame.new({num: [6]}, index: ['a']) }
   end
+
+  context 'when dataframe tuples contain nils in mismatching positions' do
+
+    let(:df){
+      Daru::DataFrame.new(
+        {
+          'string1' => ["Color", "Color", "Color", "Color", nil, "Color", "Color", " Black and White"],
+          'string2' => ["Test", "test2", nil, "test3", nil, "test", "test3", "test5"],
+          'num' => [1, nil, 3, 4, 5, 6, 7, nil]
+        }
+      )
+    }
+
+    it 'groups by without errors' do
+      expect { df.group_by(df.vectors.map(&:to_s)) }.to_not raise_error(ArgumentError)
+    end
```

Comment:
Let's add a bit more specs :)
1. For results (not just "it doesn't fail")
2. For different cases for sorting, it can be buggy (for ex., I wrote that from the top of the head, and not sure if it should be `1` and `-1` in the first two lines or vice versa).

Reply:
@zverok I didn't bother with testing the actual sorting too much because I think that's beyond the scope of this change - there are plenty of other specs that test the actual sorting part for all the other scenarios that break when you change the current behaviour).